packages/android-client/FishjamClient/src/main/java/com/fishjamcloud/client/ReconnectionManager.kt

```diff
@@ -0,0 +1,78 @@
+package com.fishjamcloud.client
+
+import com.fishjamcloud.client.models.ReconnectConfig
+import timber.log.Timber
+import java.util.Timer
+import kotlin.concurrent.schedule
+
+interface ReconnectionManagerListener {
+  fun onReconnectionStarted() {
+    Timber.i("Reconnection started")
+  }
+
+  fun onReconnected() {
+    Timber.i("Reconnected successfully")
+  }
+
+  fun onReconnectionRetriesLimitReached() {
+    Timber.e("Reconnection retries limit reached")
+  }
+}
+
+enum class ReconnectionStatus {
+  IDLE,
+  RECONNECTING,
+  ERROR,
+  WAITING
+}
+
+internal class ReconnectionManager(
+  private var reconnectConfig: ReconnectConfig = ReconnectConfig(),
+  private val connect: () -> Unit
+) {
+  private val listeners = mutableListOf<ReconnectionManagerListener>()
+  private var reconnectAttempts = 0
+  private var reconnectionStatus = ReconnectionStatus.IDLE
+
+  fun onDisconnected() {
+    if (reconnectAttempts >= reconnectConfig.maxAttempts) {
+      reconnectionStatus = ReconnectionStatus.ERROR
+      listeners.forEach { it.onReconnectionRetriesLimitReached() }
+      return
+    }
+
+    if (reconnectionStatus == ReconnectionStatus.WAITING) {
+      return
+    }
+
+    if (reconnectionStatus != ReconnectionStatus.RECONNECTING) {
+      reconnectionStatus = ReconnectionStatus.RECONNECTING
+      listeners.forEach { it.onReconnectionStarted() }
+    }
+    val delay = reconnectConfig.initialDelayMs + reconnectAttempts * reconnectConfig.delayMs
```

Comment:
lets put initialDelay = delay = 100, reconnect attempts = 5
Now process looks of connection looks like:
Failure occurs
1st attempt to reconnect, connect is called after 100 + 0 * 100 
Failure occurs again
2 attempt to reconnect, connect is called after 100 + 1 * 100
...
5 attempt to reconnect, connect is called after 100 + 4 * 100
total time of waiting for reconnecting is 1500 
Description says it is exponential backoff, while it is linear because we increase delay by delayMs. 
To make it exponential backoff it should be sth like:
`val delay = min(maxDelay, pow(2, reconnectAttempts) * delay)` 
I think that it should be documented somewhere how delay is calculated before merging it to main, and it should be decides is it some linear delay or exponential backoff.

Reply:
https://github.com/fishjam-cloud/web-client-sdk/blame/main/packages/ts-client/src/reconnection.ts#L144 I followed the web client
So I'll change the name to "linear backoff"?

Reply:
What are pros of linear compared to exponential?
I heard that exponential backoff is better. Mayby lets implement it? 
Not sure which one is better but in most cases it is used.

Reply:
I also don't know which one is better, probably it doesn't matter because user will restart the app after ~10 seconds anyway, so I made it to be the same as web for consistency